src/orquestra/sdk/_base/cli/_dorq/_dumpers.py

```diff
@@ -82,27 +82,39 @@ class LogsDumper:
     Writes logs to files.
     """
 
+    @staticmethod
+    def _get_logs_file(
+        dir_path: Path, wf_run_id: WorkflowRunId, suffix: t.Optional[str] = None
+    ) -> Path:
+        dir_path.mkdir(parents=True, exist_ok=True)
+        if suffix:
+            return dir_path / f"{wf_run_id}_{suffix.lower().replace(' ', '_')}.log"
+        return dir_path / f"{wf_run_id}.log"
+
     def dump(
         self,
-        logs: t.Mapping[TaskInvocationId, t.Sequence[str]],
+        logs: t.Union[t.Mapping[TaskInvocationId, t.Sequence[str]], t.Sequence[str]],
         wf_run_id: WorkflowRunId,
         dir_path: Path,
+        log_type: t.Optional[str] = None,
     ) -> Path:
         """
-        Save logs from wf into a file
+        Save logs from wf into a file.
 
         Creates missing directories. Generates filenames based on ``wf_run_id``
         No standard errors are expected to be raised.
         """
-        dir_path.mkdir(parents=True, exist_ok=True)
-
-        logs_file = dir_path / f"{wf_run_id}.log"
+        logs_file = self._get_logs_file(dir_path, wf_run_id, suffix=log_type)
 
         with logs_file.open("w") as f:
-            for task_invocation in logs:
-                f.write(f"Logs for task invocation: {task_invocation}:\n\n")
-                for log in logs[task_invocation]:
+            if isinstance(logs, t.Mapping):
```

Comment:
**suggestion**: I think this works, but in general `isinstance(foo, t.Bar)` might not always work because some protocols are not runtime-checkable. The union type for `logs` seems to only be used with `isinstance()` checks. Looks like this use case can be solved in a less fragile way.

Instead of a single method with a union, I'd suggest writing a separate method with non-union argument types. It doesn't have to be copy+paste duplication, the common bits can be extracted to private functions or methods.

Reply:
As an alternative:

`dump`'s signature remains unchanged, but it calls out to a separate method (`_construct_output_log_lines`) to unify the potential log classes. `_construct_output_log_lines` is a singledispatcher with overloads for lists and dicts.

My reasoning for doing this rather than splitting dump is that I want to keep the handling of logs outside of the presenter/dumper agnostic to the type, as this a) makes it much easier to parametrize accessing logs, and b) keeps the differences about how the logs are presented/formatted contained to the classes doing that presentation.